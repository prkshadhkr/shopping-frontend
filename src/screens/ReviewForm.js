import React, { useState } from 'react';

const ReviewForm = ({ submitReview }) => {
  const initialValue = {
    title: "",
    rating: "",
    comment: ""
  }
  const [formData, setFormData] = useState(initialValue);

  const handleSubmit = e => {
    e.preventDefault();
    submitReview(formData.title, formData.rating, formData.comment);
    setFormData(initialValue);
  }

  const handleChange = e => {
    const { name, value } = e.target;
    setFormData(d => ({
      ...d,
      [name]: value
    }));
  }

  return (
		<div>
    <div className="card">
      <div className="card-body">
        <form onSubmit={handleSubmit} >
          <div className="form-group row">
						<label htmlFor="for-title" className="col-sm-2">Title</label>
							<div className="col-sm-10">
								<input 
									id="for-title"
									name="title"
									value={formData.title}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>
          <div className="form-group row">
						<label htmlFor="for-rating" className="col-sm-2">Rating</label>
							<div className="col-sm-10">
								<input 
									id="for-rating"
									type="number"
									name="rating"
									value={formData.rating}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>
          <div className="form-group row">
						<label htmlFor="for-comment" className="col-sm-2">Comment</label>
							<div className="col-sm-10">
								<textarea 
									id="for-comment"
									name="comment"
									value={formData.comment}
									onChange={handleChange}
									className="form-control"/>
							</div>
					</div>

          <button 
						type="submit"
						className="btn btn-outline-primary float-right"
					>
						  Add review
					</button>
        </form>
      </div>
    </div>
		</div>
  )
}

export default ReviewForm;