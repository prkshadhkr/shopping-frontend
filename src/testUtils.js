import React from  'react';
import UserContext from './user/UserContext';

/** following is for mock store */
import configureStore from 'redux-mock-store';
import thunk from 'redux-thunk';
import axios from 'axios'
import MockAdapter from 'axios-mock-adapter';

const mock = new MockAdapter(axios);
const middlewares = [thunk];
const mockStore = configureStore(middlewares);
const initialState = {};
const makeMockStore = (state = {}) => {
  return mockStore ({
    ...initialState,
    ...state
  })
}

/** create a demouser */
const demoUser = {
  username: "testing",
  first_name: "testing",
  last_name: "testing",
  email: "testing@gmail.com",
  password: "testing",
  is_admin: false
};

/**
 * user is destructured from userContext in Profilejs as
 * const { user, setUser } = useContext(UserContext);
 */
const UserProvider = ({ children, user = demoUser }) => (
  <UserContext.Provider value={{user}}>
    {children}
  </UserContext.Provider>
)

export { mock, makeMockStore, UserProvider };