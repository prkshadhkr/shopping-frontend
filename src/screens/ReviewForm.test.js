import React from 'react';
import { render } from '@testing-library/react';
import ReviewForm from './ReviewForm';
import { MemoryRouter } from 'react-router';
import { Provider } from "react-redux";
import { makeMockStore } from '../testUtils';

const store = makeMockStore();

it ('renders without crashing', function() {
  render(
    <MemoryRouter>
      <Provider store={store}>
        <ReviewForm />
      </Provider>
    </MemoryRouter>
  )
});

it ('matches snapshot', function() {
  const { asFragment } = render (
    <MemoryRouter>
      <Provider store={store}>
        <ReviewForm />
      </Provider>
    </MemoryRouter>
  );
  expect(asFragment()).toMatchSnapshot();
})