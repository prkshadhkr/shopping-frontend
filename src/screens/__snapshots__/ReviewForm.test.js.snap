// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div>
    <div
      class="card"
    >
      <div
        class="card-body"
      >
        <form>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2"
              for="for-title"
            >
              Title
            </label>
            <div
              class="col-sm-10"
            >
              <input
                class="form-control"
                id="for-title"
                name="title"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2"
              for="for-rating"
            >
              Rating
            </label>
            <div
              class="col-sm-10"
            >
              <input
                class="form-control"
                id="for-rating"
                name="rating"
                type="number"
                value=""
              />
            </div>
          </div>
          <div
            class="form-group row"
          >
            <label
              class="col-sm-2"
              for="for-comment"
            >
              Comment
            </label>
            <div
              class="col-sm-10"
            >
              <textarea
                class="form-control"
                id="for-comment"
                name="comment"
              />
            </div>
          </div>
          <button
            class="btn btn-outline-primary float-right"
            type="submit"
          >
            Add review
          </button>
        </form>
      </div>
    </div>
  </div>
</DocumentFragment>
`;
